db/database_test.go

```diff
@@ -1355,6 +1354,11 @@ func TestPostWithUserSpecialProperty(t *testing.T) {
 }
 
 func TestIncrRetrySuccess(t *testing.T) {
+
+	if !base.UnitTestUrlIsWalrus() {
+		t.Skip("Test only works against Walrus, due to incrWithRetry being short-circuited in the LeakyBucket + CouchbaseBucketGoCB case")
+	}
```

Comment:
Can you provide some details on what's changed to make this Walrus-only?

Reply:
I originally noticed this in https://github.com/couchbase/sync_gateway/pull/3613#issuecomment-393618626 and I just added some comments in 8cc96f061553f9b217d85e008d40ef39b4973b79 with my understanding of why that test fails when using a gocb bucket.